src/benchmarks/micro/corefx/System.Runtime/Perf.Enum.cs

```diff
@@ -20,19 +22,534 @@ public enum Colors
+
+        [Benchmark]
+        //[Arguments((BigContiguousEnum)36)] // BenchmarkDotNet Issue #1020
+        //[Arguments((SmallContiguousEnum)3)]
+        //public string ToStringUndefined(Enum value) => value.ToString();
+        public string ToStringUndefined() => ((BigContiguousEnum)36).ToString();
+
+        [Benchmark]
+        //[Arguments(Colors.Red | Colors.Orange | Colors.Yellow | Colors.Green | Colors.Blue)] // BenchmarkDotNet Issue #1020
+        //public string ToStringFlags5(Enum value) => value.ToString();
+        public string ToStringFlags5() => (Colors.Red | Colors.Orange | Colors.Yellow | Colors.Green | Colors.Blue).ToString();
+
+        [Benchmark]
+        //[Arguments(Colors.Orange | Colors.Yellow | Colors.Green)] // BenchmarkDotNet Issue #1020
+        //public string ToStringFlags3(Enum value) => value.ToString();
+        public string ToStringFlags3() => (Colors.Orange | Colors.Yellow | Colors.Green).ToString();
+
+        [Benchmark]
+        [Arguments(Colors.Red)]
+        public string ToStringFlags1(Enum value) => value.ToString();
+
+        [Benchmark]
+        //[Arguments((Colors)0x20)] // BenchmarkDotNet Issue #1020
+        //public string ToStringFlagsInvalidFlagCombination(Enum value) => value.ToString();
+        public string ToStringFlagsInvalidFlagCombination() => ((Colors)0x20).ToString();
+
+        [Benchmark]
+        [Arguments(BigContiguousEnum.C, "D")]
+        [Arguments(BigContiguousEnum.C, "X")]
+        [Arguments(BigContiguousEnum.C, "G")]
+        //[Arguments(Colors.Green | Colors.Yellow | Colors.Orange, "F")] // BenchmarkDotNet Issue #1020
+        public string ToStringFormat(Enum value, string format) => value.ToString(format);
 
         [Benchmark]
-        public bool HasFlag() => _greenAndRed.HasFlag(Colors.Green);
+        public string ToStringFormatF() => (Colors.Green | Colors.Yellow | Colors.Orange).ToString("F");
+        #endregion
+
+        #region Format
+        [Benchmark]
+        [Arguments(typeof(BigContiguousEnum), BigContiguousEnum.C, "D")]
+        [Arguments(typeof(BigContiguousEnum), BigContiguousEnum.C, "X")]
+        [Arguments(typeof(BigContiguousEnum), BigContiguousEnum.C, "G")]
+        //[Arguments(typeof(Colors), Colors.Green | Colors.Yellow | Colors.Orange, "F")] // BenchmarkDotNet Issue #1020
+        public string Format(Type enumType, Enum value, string format) => Enum.Format(enumType, value, format);
+
+        [Benchmark]
+        [Arguments(typeof(Colors))]
+        public string FormatF(Type enumType) => Enum.Format(enumType, Colors.Green | Colors.Yellow | Colors.Orange, "F");
+        #endregion
+
+        [Benchmark]
+        [Arguments(BigContiguousEnum.C, BigContiguousEnum.E)]
+        public int CompareToEnum(Enum value, object target) => value.CompareTo(target);
+
+        [Benchmark]
+        [Arguments(BigContiguousEnum.C)]
+        public int CompareToNull(Enum value) => value.CompareTo(null);
```

Comment:
what time does this benchmark report? Is it `0ns`?

Reply:
`~2ns`